content/blog/2021/06/kwctl-intro-for-kubernetes-administrators.md

```diff
@@ -0,0 +1,277 @@
+        -r req_pod_with_container_with_capabilities_added.json \
+        --settings-json '{"default_add_capabilities": ["AUDIT_READ"]}'
+```
+
+The command will fail with the following output:
+```shell 
+{"valid":false,"message":"These capabilities cannot be added by default because they are not allowed: {\"AUDIT_READ\"}"}
+Error: Provided settings are not valid: Some("These capabilities cannot be added by default because they are not allowed: {\"AUDIT_READ\"}")
+```
+
+Nice, the author of the policy implemented some validation of the settings
+provided by the end users!
+The error message points Alice in the right direction: the `AUDIT_READ` capability
+cannot be added to all the containers unless it's also on the list of the
+allowed capabilities.
+
+Alice runs the policy again, this time with a different configuration:
+
+```shell
+$ kwctl run registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.4 \
+      -r req_pod_with_container_with_capabilities_added.json \
+      --settings-json '{"default_add_capabilities": ["AUDIT_READ"], "allowed_capabilities": ["AUDIT_READ"]}'
+``` 
+
+This time the configuration is correct, but the request is rejected with
+the following explanation:
+
+```json
+{
+  "uid": "1299d386-525b-4032-98ae-1949f69f9cfc",
+  "allowed": false,
+  "status": {
+    "message": "PSP capabilities policies doesn't allow these capabilities to be added: {\"NET_ADMIN\", \"SYS_TIME\"}"
+  }
+}
+```
+
+That's actually a correct behaviour, Alice forgot to allow the capabilities
+the container is already adding to itself.
+
+Alice now runs the kwctl command one last time:
+
+```shell
+$ kwctl run registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.4 \
+        -r req_pod_with_container_with_capabilities_added.json \
+        --settings-json '{"default_add_capabilities": ["AUDIT_READ"], "allowed_capabilities": ["AUDIT_READ", "NET_ADMIN", "SYS_TIME"]}'
+```
+
+This time the policy accepts the incoming request, plus it mutates it.
+This is the output returned:
+
+```json
+{
+  "uid": "1299d386-525b-4032-98ae-1949f69f9cfc",
+  "allowed": true,
+  "patchType": "JSONPatch",
+  "patch": "W3sib3AiOiJhZGQiLCJwYXRoIjoiL3NwZWMvY29udGFpbmVycy8wL3NlY3VyaXR5Q29udGV4dC9jYXBhYmlsaXRpZXMvYWRkLzIiLCJ2YWx1ZSI6IkFVRElUX1JFQUQifV0="
+}
+```
+
+Alice can see what the policy is going to change by looking at the contents of
+the `patch` attribute.
+As requested by Kubernetes, the patch operation is encoded using `base64`:
+
+```shell
+$ kwctl run registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.4 \
+        -r req_pod_with_container_with_capabilities_added.json \
+        --settings-json '{"default_add_capabilities": ["AUDIT_READ"], "allowed_capabilities": ["AUDIT_READ", "NET_ADMIN", "SYS_TIME"]}' \
+  | jq -r .patch | base64 -d
+```
+
+This produces the following output:
+```json
+[{"op":"add","path":"/spec/containers/0/securityContext/capabilities/add/2","value":"AUDIT_READ"}]
```

Comment:
I think a short statement explaining what  this JSONPatch does would be great to finish this section.